### DISCUSSION TITLE
DB migration in the style of .NET Entity Framework (EF Core)

### DISCUSSION THREAD

**Discussion by frvdh** [sequelize/sequelize#15365] [Help & Questions] [answered]
Hello,

I've been making a PostgreSQL DB and using Sequelize as ORM. After reading and experimenting with the documentation on DB migrations, an idea came up: would it be possible to update the DB model and do a migration by using CLI commands and simply changing the model definition? (like is done in the .NET Entity Framework DB migration: https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/?tabs=dotnet-core-cli ) 

 E.g. going from this:
`const User = sequelize.define('User', {
  id: {
    type: DataTypes.STRING,
    primaryKey: true,
    allowNull: false,
  },
  role: {
    type: DataTypes.STRING,
  },
});`

To this:
`const User = sequelize.define( 'User',
    {
      id: {
        type: DataTypes.STRING,
        primaryKey: true,
        allowNull: false,
      },
      role: {
        type: DataTypes.STRING,
      },
      username: {
        type: DataTypes.STRING,
      },
    },
);`

and then running a command similar to (from the EF Core docs): 
`dotnet ef migrations add AddBlogCreatedTimestamp`

If this isn't possible, then I'll just continue using the migration JS files like shown in the Sequelize DB migratrion docs but I thought it would also be a cool feature to easily make DB migrations if it didn't exist in the Sequelize lib yet.

Thanks for your answer!

**Comment by ephys** [comment, marked as answer]
Unfortunately we don't have such a feature :(
I'd love to see it being added, and have [a draft of an analysis of how it could work](https://gist.github.com/ephys/59968ad1c7446734f97eafa6f37e7bfd), but it's not a priority for us when compared to other features and bugs we currently have on our backlog

**Reply by frvdh** [reply]
Hi Ephys, OK I understand. Thanks for your answer!
